### PR TITLE
Update MUSIC tutorial

### LINKED ISSUE #1495: MUSIC tutorial on ReadTheDocs needs updating
**Describe the bug**
The MUSIC tutorial on ReadTheDocs is outdated and needs to be updated to fit with NEST 3.0. Especially the [MUSIC with SLI](https://nest-simulator.readthedocs.io/en/latest/tutorials/music_tutorial/music_tutorial_sli.html) documentation, which references `subnet` and outdated functions like `LayoutNetwork`. The other pages are outdated as well though.

**Screenshots**
Looking very quickly at the first Python example in [Connect two NEST simulations using MUSIC](https://nest-simulator.readthedocs.io/en/latest/tutorials/music_tutorial/music_tutorial_2.html) you can see some of the outdated code:

MUSIC_tutorial

### PR DESCRIPTION
**Pull request by stinebuu**
This PR updates the MUSIC tutorial for ReadTheDocs. I have updated the code snippets to work with NEST 3.0 and updated the text in a few places to fit with the changes as well.

`music_tutorial_sli.rst` was not previously part of the tutorial on ReadTheDocs, you could search for it and find it, but it was not part of the index. I don't know if this was deliberate? I placed it between part 2 and 3 of the tutorial, because part 2 is about MUSIC connected to NEST, and part 3+4 are about MUSIC C++ and Python interfaces. But let me know if I should remove it again.

I am also wondering if part 2 and 3 are relevant for the NEST documentation? 
I have read through these quickly, but it would be great if someone else could have a closer look.

Also, I am not getting MUSIC to work at the moment, so I have not tested that the code snippets work properly. I know the NEST code runs (until simulation, where I get a MUSIC error), but I have not been able to test that the MUSIC configuration file works. This is particularly important for lines [102-114 in `music_tutorial_sli.rst`](https://github.com/nest/nest-simulator/pull/1552/files#diff-5e27bcba41f3596e8a5cdaa64492a5a2R102-R114), because I added the code for the configuration. So I suggest @jougs should have a close look here.

This fixes #1495 .

### CODE REVIEW

File: doc/tutorials/music_tutorial/music_tutorial_2.rst [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 
 Let’s look at an example of two NEST simulations connected through
 MUSIC. We’ll implement the simple network in :numref:`neuronmusic3`
 from :doc:`the introduction to this tutorial <music_tutorial_1>`.
 
 We need a sending process, a receiving process and a MUSIC
-configuration file:
+configuration file.
+
+To try out the example, save the following sending process code in a python file
```
Comment (jougs) [inline doc/tutorials/music_tutorial/music_tutorial_2.rst:11]:
```suggestion
To try out the example, save the following sending process code in a Python file
```

File: doc/tutorials/music_tutorial/music_tutorial_1.rst [hunk 2]
```diff
@@ -47,18 +47,16 @@ constructs.
...
+~~~~~~~~~~~~~~~~~~
 
 A NEST network consists of three types of elements: neurons, devices,
 and connections between them.
 
 Neurons are the basic building blocks, and in NEST they are generally
 spiking point neuron models. Devices are supporting units that for
 instance generate inputs to neurons or record data from them. The
-Poisson spike generator, the spike detector recording device and the
+poisson spike generator, the spike detector recording device, and the
```
Comment (jougs) [inline doc/tutorials/music_tutorial/music_tutorial_1.rst:59]:
```suggestion
Poisson spike generator, the spike detector recording device, and the
```

The distribution is called after a french mathematician, so the name is written with upper-case starting letter. So either write it like it's supposed to be , or say `poisson_generator` and `spike_detector` (probably both in double backticks to get them formatted and linked properly). I'm fine with both and provide one of them for a seamless clicking review response experience ;-)

File: doc/tutorials/music_tutorial/music_tutorial_2.rst [hunk 3]
```diff
@@ -17,16 +20,15 @@ configuration file:
     import nest
     nest.SetKernelStatus({"overwrite_files": True})
 
-    neurons = nest.Create('iaf_psc_alpha', 2, [{'I_e': 400.0}, {'I_e': 405.0}])
+    neurons = nest.Create('iaf_psc_alpha', 2, params={'I_e': [400.0, 405.0]})
```
Comment (jougs) [inline doc/tutorials/music_tutorial/music_tutorial_2.rst:23]:
```suggestion
    neurons = nest.Create('iaf_psc_alpha', 2, {'I_e': [400.0, 405.0]})
```

There's also no keywords for the first two arguments

File: doc/tutorials/music_tutorial/music_tutorial_2.rst [hunk 4]
```diff
@@ -17,16 +20,15 @@ configuration file:
...
 
-    music_out = nest.Create('music_event_out_proxy', 1,
-        params = {'port_name':'p_out'})
+    music_out = nest.Create('music_event_out_proxy', 1, params={'port_name':'p_out'})
```
Comment (jougs) [inline doc/tutorials/music_tutorial/music_tutorial_2.rst:25]:
```suggestion
    music_out = nest.Create('music_event_out_proxy', 1, {'port_name':'p_out'})
```

File: doc/tutorials/music_tutorial/music_tutorial_2.rst [hunk 5]
```diff
@@ -61,18 +65,16 @@ simulate for one second.
     import nest
     nest.SetKernelStatus({"overwrite_files": True})
 
-    music_in = nest.Create("music_event_in_proxy", 2,
-        params = {'port_name': 'p_in'})
+    music_in = nest.Create("music_event_in_proxy", 2, params={'port_name': 'p_in'})
```
Comment (jougs) [inline doc/tutorials/music_tutorial/music_tutorial_2.rst:68]:
```suggestion
    music_in = nest.Create("music_event_in_proxy", 2, {'port_name': 'p_in'})
```

File: doc/tutorials/music_tutorial/music_tutorial_sli.rst [hunk 6]
```diff
@@ -14,56 +14,58 @@ the scope of this tutorial. The code follows the same structure as the
 other examples, and should be straightforward to follow. But we will
 give a few pointers for how to connect things with MUSIC.
 
+The sli version of the sending process file from
```
Comment (jougs) [inline doc/tutorials/music_tutorial/music_tutorial_sli.rst:17]:
```suggestion
The SLI version of the sending process file from
```

File: doc/tutorials/music_tutorial/music_tutorial_4.rst [hunk 7]
```diff
@@ -100,7 +100,7 @@ C++.
 
 As the ``pymusic`` bindings are still quite new the
 documentation is still lagging behind. This quick introduction should
-nevertheless bee enough for you to get going with the bindings. And
+nevertheless be enough for you to get going with the bindings. And
 should you need further help, the authors are only an email away.
```
[the remark this answers is not in the document]
Comment (stinebuu) [reply doc/tutorials/music_tutorial/music_tutorial_4.rst:104]: I don't know, do you want me to remove the sentence or put in nest-user e-mail?

[the remark this answers is not in the document]
Comment (sarakonradi) [reply doc/tutorials/music_tutorial/music_tutorial_4.rst:104]: The NEST Users email would be great!

Comment (sarakonradi) [inline doc/tutorials/music_tutorial/music_tutorial_4.rst:104]:
```suggestion
nevertheless be enough for you to get going with the bindings. Feel
free to ask our `Mailing List <https://www.nest-initiative.org/mailinglist/>`_ 
if you need further help.
```

Comment (stinebuu) [reply doc/tutorials/music_tutorial/music_tutorial_4.rst]: Thanks @sarakonradi! Time got away again.

**Pull request merged**